examples/static-file-server/src/main.rs

```diff
@@ -86,6 +88,24 @@ fn using_serve_dir_only_from_root_via_fallback() -> Router {
         .fallback_service(serve_dir)
 }
 
+fn using_serve_dir_with_handler_as_service() -> Router {
+    // you can convert handler function to service
+    async fn handler_404() -> impl IntoResponse {
+        (StatusCode::NOT_FOUND, "Not found")
+    }
+    
+    let service = handler_404
+        .into_service()
+        .map_err(|_| io::Error::from(io::ErrorKind::Other));
```

Comment:
Does this work?

```suggestion
        .map_err(|err| match err {}));
```

Reply:
No. With this it throws error:
```
error[E0277]: the trait bound `std::io::Error: From<()>` is not satisfied
   --> static-file-server/src/main.rs:103:21
    |
103 |     let serve_dir = get_service(serve_dir).handle_error(handle_error);
    |                     ^^^^^^^^^^^ the trait `From<()>` is not implemented for `std::io::Error`
```

Reply:
Ah right. This works:

```rust
        .map_err(|err| -> std::io::Error { match err {} });
```

I think we should use that because it makes it clear that we never actually create an `io::Error`.

Reply:
yeah, this one works. Commited